src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/import.py

```diff
@@ -36,53 +36,44 @@ def acr_import(cmd,
-                source_registry = registry.id
-        source = ImportSource(resource_id=source_registry, source_image=source_image)
+                source = ImportSource(resource_id=registry.id, source_image=source_image)
+            else:
+                raise CLIError(SOURCE_REGISTRY_NOT_FOUND)
     else:
-        source_registry_login_server = source_image[:slash]
-        source_image = source_image[slash + 1:]
-        if not source_image or not source_registry_login_server:
-            raise CLIError(INVALID_SOURCE_IMAGE)
-        registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry_login_server)
-        if registry:
-            if source_registry and \
-               source_registry.lower() != registry.id.lower() and \
-               source_registry.lower() != registry.name.lower() and \
-               source_registry.lower() != registry.login_server.lower():
-                raise CLIError(REGISTRY_MISMATCH)
-            source = ImportSource(resource_id=registry.id,
-                                  source_image=source_image)
-        else:
-            if source_registry_password:
-                ImportSourceCredentials = cmd.get_models('ImportSourceCredentials')
-                if source_registry_username:
-                    source = ImportSource(registry_uri=source_registry_login_server,
+        slash = source_image.find('/')
+        if slash > 0:
+            registry_uri = source_image[:slash]
+            dot = registry_uri.find('.')
+            if dot > 0:
+                source_image = source_image[slash + 1:]
+                if source_registry_password:
+                    ImportSourceCredentials = cmd.get_models('ImportSourceCredentials')
+                    source = ImportSource(registry_uri=registry_uri,
                                           source_image=source_image,
                                           credentials=ImportSourceCredentials(password=source_registry_password,
                                                                               username=source_registry_username))
                 else:
-                    source = ImportSource(registry_uri=source_registry_login_server,
-                                          source_image=source_image,
-                                          credentials=ImportSourceCredentials(password=source_registry_password))
+                    registry = get_registry_from_name_or_login_server(cmd.cli_ctx, registry_uri)
+                    if registry:
+                        source = ImportSource(resource_id=registry.id, source_image=source_image)
+                    else:
+                        source = ImportSource(registry_uri=registry_uri, source_image=source_image)
             else:
-                source = ImportSource(registry_uri=source_registry_login_server, source_image=source_image)
+                raise CLIError(LOGIN_SERVER_NOT_VALID)
+        else:
+            raise CLIError(SOURCE_NOT_FOUND)
```

Comment:
The message is a bit off here, you probably can just reuse the help message for `--source`